File: src/my_project/easy_problems/from51to100/count_items_matching_rule.py
from typing import List

class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:

        # initialize variables
        temp_index = None
        count = 0

        if ruleKey == "type":
            temp_index = 0
        if ruleKey == "color":
            temp_index =1
        if ruleKey == 'name':
            temp_index = 2


        for item in items:
            if item[temp_index] == ruleValue:
                count += 1





        return count


solution = Solution()
print(solution.countMatches(items = [["phone","blue","pixel"],["computer","silver","phone"],["phone","gold","iphone"]], ruleKey = "type", ruleValue = "phone"))

File: src/my_java_projects/src/easy_problems/from_1_to_50/CountItemsMatchingRule.java
package easy_problems.from_1_to_50;

import java.util.List;

public class CountItemsMatchingRule implements Numbers{
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {

        int tempIndex = -1;
        int solution = 0;

        if (ruleKey.equals("type") ){
            tempIndex = 0;
        }
        else if (ruleKey.equals("color")){
            tempIndex = 1;
        }
        else if (ruleKey.equals("name")){
            tempIndex = 2;
        }

        for (List<String> item: items){

            if (item.get(tempIndex).equals(ruleValue)){
                solution += 1;
            }
        }

        return solution;
    }

    @Override
    public void returnNumbers() {
        System.out.println("This is a class that returns the counting number of matches according to a specific criterion");
    }
}
